scipy/signal/filter_design.py

```diff
@@ -1033,6 +1033,33 @@ def zpk2sos(z, p, k, pairing='nearest'):
     return sos
 
 
+def _align_nums(nums):
+    """
+    Given an array of numerator coefficient arrays [[a_1, a_2,...,
+    a_n],..., [b_1, b_2,..., b_m]], this function pads shorter numerator
+    arrays with zero's so that all numerators have the same length. Such
+    alignment is necessary for functions like 'tf2ss', which needs the
+    alignment when dealing with SIMO transfer functions.
+    """
+    try:
+        nums = asarray(nums)
+
+        if not np.issubdtype(nums.dtype, np.number):
+            raise ValueError("dtype of numerator is non-numeric")
+
+        return nums
+
+    except ValueError:
```

Comment:
You are correct in believing that it comes from the `asarray(nums)` call.  A potential `ValueError` can be thrown in case like this <a href="https://github.com/gfyoung/scipy/blob/tf2ss_matrix_squeeze/scipy/signal/tests/test_ltisys.py#L325">one</a>.  If you try calling `asarray` on the numerator array (i.e. `tf[0]`), you will get a `ValueError` saying that you tried to set an array element with a sequence.